File: sis-kitnet/src/app/core/model/dto/controleLancamento/controleLancamentoPostDTO.ts
import { StatusDTO } from './statusDTO';
import { ValorRegraDTO } from './valorRegraDTO';
import { ValorId } from './valorId';
import { InquilinoId } from './inquilinoId';
import { ApartamentoId } from './apartamentoId';

export interface ControleLancamentoPostDTO {
  dataEntrada: string;
  dataPagamento?: string;
  observacao?: string;
  status?: StatusDTO;
  valores?: ValorRegraDTO;
  valor: ValorId;
  inquilino: InquilinoId;
  apartamento: ApartamentoId;
}

File: kitnet/src/main/java/kitnet/com/api/dto/controleLancamento/ControleLancamentoPostDTO.java
package kitnet.com.api.dto.controleLancamento;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import kitnet.com.api.dto.apartamento.ApartamentoId;
import kitnet.com.api.dto.inquilino.InquilinoId;
import kitnet.com.api.dto.valor.ValorId;
import java.time.LocalDate;

public record ControleLancamentoPostDTO(
    @NotNull(message = "Data de entrada é obrigatória")
    LocalDate dataEntrada,
    LocalDate dataPagamento,
    @Size(max = 500, message = "Observação deve ter no máximo 500 caracteres")
    String observacao,
    @Valid
    StatusDTO status,
    @Valid
    ValorRegraDTO valores,
    @NotNull(message = "Valor é obrigatório")
    ValorId valor,
    @NotNull(message = "Inquilino é obrigatório")
    InquilinoId inquilino,
    @NotNull(message = "Apartamento é obrigatório")
    ApartamentoId apartamento
) {}
